contracts/StorageContract.sol

```diff
@@ -9,10 +9,61 @@ import "./RandaoLib.sol";
  * EthStorage L1 Contract with Decentralized KV Interface and Proof of Storage Verification.
  */
 abstract contract StorageContract is DecentralizedKV {
+    struct Config {
+        uint256 maxKvSizeBits;
+        uint256 shardSizeBits;
+        uint256 randomChecks;
+        uint256 minimumDiff;
+        uint256 cutoff;
+        uint256 diffAdjDivisor;
+        uint256 treasuryShare; // 10000 = 1.0
+    }
+
+    uint256 public constant sampleSizeBits = 5; // 32 bytes per sample
+
+    uint256 public immutable maxKvSizeBits;
+    uint256 public immutable shardSizeBits;
+    uint256 public immutable shardEntryBits;
+    uint256 public immutable sampleLenBits;
+    uint256 public immutable randomChecks;
+    uint256 public immutable minimumDiff;
```

Comment:
nonceLimit, minimumDiff and prepaidAmount are not related to the gas cost of putBlob, and I think we can change them to public variables and add only owner setter func.